lib/Runtime/Types/ES5ArrayTypeHandler.cpp

```diff
@@ -271,25 +272,23 @@ namespace Js
             doneConversion = true;
 #endif
         }
-        __finally
+        catch(...)
```

Comment:
It just carries the functionality as is. Previous logic was defined as to run cleanup part no matter what and do not catch anything. Current version does the same. catch everything, cleanup. and throw. So, IMHO there was no point for hosting exception info inside the scope stack.

PR is updated with Js::ExceptionBase.